lib/jasmine/jasmine.ts

```diff
@@ -1,32 +1,97 @@
-        || Zone.current.getZoneWith('ProxyZoneSpec')
-        || originalZone;
-    zone.scheduleMicroTask('jasmineCleanStack', fn);
+  const ambientZone = Zone.current;
+  // Sync zone will throw on any async operations. 
+  // Will be used when executing contents of describe.
+  const syncZone = ambientZone.fork(new SyncTestZoneSpec('jasmine.describe'));
+
+  // This is the zone which will be used for running individual tests.
+  // It will be a proxy zone, so that the tests function can retroactivly install
+  // different zones. 
+  let testProxyZone: Zone = null;
+
+  // Monkey patch all of the jasmine DSL so that each function runs in appropriate zone.
+  const jasmineEnv = jasmine.getEnv();
+  ['desribe', 'xdescribe', 'fdescribe'].forEach((methodName) => {
+    var delegate: Function = jasmineEnv[methodName];
+    jasmineEnv[methodName] = function(description: string, specDefinitions: Function) {
+      return delegate.call(this, description,  wrapDescribeInZone(specDefinitions));
+    } 
+  });
+  ['it', 'xit', 'fit'].forEach((methodName) => {
+    var delegate: Function = jasmineEnv[methodName];
+    jasmineEnv[methodName] = function(description: string, specDefinitions: Function) {
+      return delegate.call(this, description, wrapTestInZone(specDefinitions));
+    } 
+  });
+  ['beforeEach', 'afterEach'].forEach((methodName) => {
+    var delegate: Function = jasmineEnv[methodName];
+    jasmineEnv[methodName] = function(specDefinitions: Function) {
+      return delegate.call(this, wrapTestInZone(specDefinitions));
+    } 
+  });
+
+  function wrapDescribeInZone(fn: Function): Function {
+    // Used for executing the describe content. No async work is allowed here.
+    return function() {
+      return syncZone.run(fn, this, arguments as any as any[]);
+    }
   }
 
-  function QueueRunner(options) {
-    options.clearStack = alwaysClearStack;
-    SuperQueueRunner.call(this, options);
+  function wrapTestInZone(fn: Function): Function {
+    // Used for executing it/beforeEach/afterEach. This will run in `testProxyZone`.
+    // The `testProxyZone` will be reset by the `ZoneQueueRunner`
+    if(fn.length == 0) {
+      return function() {
+        return testProxyZone.run(fn, this);
+      }
+    } else {
+      return function(done) {
+        return testProxyZone.run(fn, this, [done]);
+      }
+    }
   }
-  QueueRunner.prototype = SuperQueueRunner.prototype;
-  return QueueRunner;
-})((<any>jasmine).QueueRunner);
+  interface QueueRunner {
+    execute(): void;
+  }
+  interface QueueRunnerAttrs {
+    queueableFns: {fn: Function}[];
+    onComplete: () => void;
+    clearStack: (fn) => void;
+    onException: (error) => void;
+    catchException: () => boolean;
+    userContext: any;
+    timeout: {setTimeout: Function, clearTimeout: Function};
+    fail: ()=> void;
+  }
+
+  const QueueRunner = (jasmine as any).QueueRunner as { new(attrs: QueueRunnerAttrs): QueueRunner };
```

Comment:
This is a standard practice in TypeScript, except that most of the times the Type is auto extracted for you. You want to be able to write: `let queueRunner: QueueRunner = new QueueRunner()` and this allows it. So I think we should keep it as is.